### PR TITLE
Sending lab test to ALIS Task

### PR DESCRIPTION
**Pull request by Mwanje**
https://app.asana.com/0/1133167201254915/1201005948152251

### CODE REVIEW

File: api/src/main/java/org/openmrs/module/ugandaemrsync/server/SyncFHIRRecord.java [hunk 1]
```diff
@@ -424,12 +424,9 @@ public void identifyNewCases(SyncFhirProfile syncFhirProfile, Date currentDate)
             List list=  Context.getAdministrationService().executeSQL("select Distinct patient_id from  orders where date_activated <= now()",true);
```
Comment (ssmusoke) [inline api/src/main/java/org/openmrs/module/ugandaemrsync/server/SyncFHIRRecord.java:424]: @Mwanje Isn't there an orders service method that can do this?

File: api/src/main/java/org/openmrs/module/ugandaemrsync/server/SyncFHIRRecord.java [hunk 2]
```diff
@@ -421,15 +423,20 @@ public void identifyNewCases(SyncFhirProfile syncFhirProfile, Date currentDate)
...
-            List<Patient> patientList=new ArrayList<>();
 
+            String input = OpenmrsUtil.firstSecondOfDay(new Date()).toString();
+            SimpleDateFormat parser = new SimpleDateFormat("EEE MMM d HH:mm:ss zzz yyyy");
+            Date date = parser.parse(input);
+
+            SimpleDateFormat formatter = new SimpleDateFormat("yyyy-MM-dd");
+            String formattedDate = formatter.format(date);
 
+            List list=  Context.getAdministrationService().executeSQL("select Distinct patient_id from  orders where date_activated >= "+ formattedDate +" and date_stopped is NULL;",true);
```
Comment (slubwama) [inline api/src/main/java/org/openmrs/module/ugandaemrsync/server/SyncFHIRRecord.java:434]: Incorporate the order Type in your query to distinct this to tests. Currently any order goes even medication orders.

Comment (Mwanje) [reply api/src/main/java/org/openmrs/module/ugandaemrsync/server/SyncFHIRRecord.java:434]: Done

File: api/src/main/java/org/openmrs/module/ugandaemrsync/server/SyncFHIRRecord.java [hunk 3]
```diff
@@ -424,14 +424,10 @@ public void identifyNewCases(SyncFhirProfile syncFhirProfile, Date currentDate)
...
-            String input = OpenmrsUtil.firstSecondOfDay(new Date()).toString();
-            SimpleDateFormat parser = new SimpleDateFormat("EEE MMM d HH:mm:ss zzz yyyy");
-            Date date = parser.parse(input);
-
             SimpleDateFormat formatter = new SimpleDateFormat("yyyy-MM-dd");
-            String formattedDate = formatter.format(date);
+            String formattedDate = formatter.format(new Date());
 
-            List list=  Context.getAdministrationService().executeSQL("select Distinct patient_id from  orders where date_activated >= "+ formattedDate +" and date_stopped is NULL;",true);
+            List list=  Context.getAdministrationService().executeSQL("select Distinct patient_id from  orders where order_type_id = 3 AND date_activated >= "+ formattedDate +" and date_stopped is NULL ;",true);
```
Comment (slubwama) [inline api/src/main/java/org/openmrs/module/ugandaemrsync/server/SyncFHIRRecord.java:430]: @Mwanje why hard code it while you can use a variable. This same filter should be used for MedicationRequest. use the orderType.getId() option.

Comment (Mwanje) [reply api/src/main/java/org/openmrs/module/ugandaemrsync/server/SyncFHIRRecord.java:430]: Done

File: api/src/main/resources/liquibase.xml [hunk 4]
```diff
@@ -755,4 +755,46 @@
...
+            <column name="url_end_point" value="http://192.168.43.53:8080/api/alisdata"/>
+            <column name="creator" value="1"/>
+            <column name="date_created" value="2021-09-21 16:20:58"/>
+            <column name="voided" value="0"/>
+            <column name="uuid" value="2f0ef683-c988-448b-b928-e3e2cf6657af"/>
+        </insert>
+    </changeSet>
+
+    <changeSet id="ugandaemrsync-20210921-1616" author="mmwanje" dbms="mysql">
```
Comment (slubwama) [inline api/src/main/resources/liquibase.xml:785]: Duplicated ID

File: api/src/main/resources/liquibase.xml [hunk 5]
```diff
@@ -755,4 +755,46 @@
         </insert>
     </changeSet>
 
+
+    <changeSet id="ugandaemrsync-20210921-1616" author="mmwanje">
```
Comment (slubwama) [inline api/src/main/resources/liquibase.xml:759]: duplicated Id

Comment (Mwanje) [reply api/src/main/resources/liquibase.xml]: Done

File: api/src/main/resources/liquibase.xml (hunk 4 above)
Comment (Mwanje) [reply api/src/main/resources/liquibase.xml]: Done

File: api/src/main/java/org/openmrs/module/ugandaemrsync/server/SyncFHIRRecord.java (hunk 1 above)
Comment (Mwanje) [reply api/src/main/java/org/openmrs/module/ugandaemrsync/server/SyncFHIRRecord.java]: @ssmusoke no method for this specific

**Comment by slubwama** [comment]
@Mwanje remove WIP from PR name

**Comment by Mwanje** [comment]
@slubwama i removed the WIP from the PR name

**Comment by slubwama** [comment]
@Mwanje resolve conflicts

**Pull request merged**